task/main.py

```diff
@@ -20,15 +20,46 @@ class DiscoverGranules:
     It will return the files if they don't exist in S3 or the md5 doesn't match
     """
 
-    def __init__(self, csv_file_name='granules.csv'):
+    def __init__(self, event):
         """
         Default values goes here
         """
-        self.csv_file_name = csv_file_name
-        self.s3_key = f"{os.getenv('s3_key_prefix', default='temp').rstrip('/')}/{self.csv_file_name}"
-        self.s3_bucket_name = os.getenv("bucket_name")
+        self.config = event.get('config')
+        self.provider = self.config.get('provider')
+        self.collection = self.config.get('collection')
+        self.discover_tf = self.collection.get('meta').get('discover_tf')
+        csv_filename = f'{self.collection["name"]}__{self.collection["version"]}.csv'
+        self.s3_key = f'{os.getenv("s3_key_prefix", default="temp").rstrip("/")}/{csv_filename}'
+        self.s3_bucket_name = os.getenv('bucket_name')
         self.session = requests.Session()
 
+    @staticmethod
```

Comment:
The below two functions should normally define return types and be assigned to a variable in the outer scope so that it is more easily readable. Functionality works but as pointed out, is inconsistent with  int, float, bool, str, tuple, and Unicode types.